### PR TITLE
Launcher command development (second try)

### PR DESCRIPTION
**Pull request by omentic**
The previous pull request was comparing against an old revision.

### CODE REVIEW

File: src/main/java/com/spartronics4915/frc2020/Constants.java [hunk 1]
```diff
@@ -80,11 +85,11 @@
...
     public static final class PanelRotator
     {
-        public static final int kBeamSensorUpID = -1;
-        public static final int kBeamSensorDownID = -1;
+        public static final int kBeamSensorUpID = 5;
+        public static final int kBeamSensorDownID = 6;
 
-        public static final int kExtendMotorID = -1;
-        public static final int kSpinMotorID = -1;
+        public static final int kExtendMotorID = 7;
```
Comment (j-james) [inline src/main/java/com/spartronics4915/frc2020/Constants.java]: The PR still isn't working properly and I don't know why.

Look [here](https://github.com/Spartronics4915/2020-InfiniteRecharge/blob/master/src/main/java/com/spartronics4915/frc2020/Constants.java#L81) for what Constants _really_ should be.

File: src/main/java/com/spartronics4915/frc2020/Constants.java [hunk 2]
```diff
@@ -80,11 +85,11 @@
 
     public static final class PanelRotator
     {
-        public static final int kBeamSensorUpID = -1;
-        public static final int kBeamSensorDownID = -1;
+        public static final int kBeamSensorUpID = 5;
+        public static final int kBeamSensorDownID = 6;
 
-        public static final int kExtendMotorID = -1;
```
Comment (omentic) [inline src/main/java/com/spartronics4915/frc2020/Constants.java:86]: The PR still isn't working properly and I don't know why.

Look [here](https://github.com/Spartronics4915/2020-InfiniteRecharge/blob/master/src/main/java/com/spartronics4915/frc2020/Constants.java#L81) for what Constants _really_ should be.

File: src/main/java/com/spartronics4915/frc2020/Constants.java (hunk 1 above)
Comment (omentic) [inline src/main/java/com/spartronics4915/frc2020/Constants.java:91]: These are incorrect - the Beam Sensor ids aren't determined yet, and the ExtendMotor and SpinMotor are 13 and 14, respectively.

File: src/main/java/com/spartronics4915/frc2020/RobotContainer.java [hunk 3]
```diff
@@ -87,8 +88,14 @@ public RobotContainer()
         mPanelRotator = new PanelRotator();
         mLED = LED.getInstance();
         mClimberCommands = new ClimberCommands();
+        mLauncherCommands = new LauncherCommands();
         mPanelRotatorCommands = new PanelRotatorCommands();
 
+        //mClimber.setDefaultCommand(mClimberCommands.new ClimberDefaultCommand(mClimber));
```
Comment (omentic) [inline src/main/java/com/spartronics4915/frc2020/RobotContainer.java:94]: Uncomment these out.

File: src/main/java/com/spartronics4915/frc2020/RobotContainer.java [hunk 4]
```diff
@@ -128,17 +135,18 @@ private void configureJoystickBindings()
 
         /* Switch Camera views
         new JoystickButton(mJoystick, 6).whenPressed(
-            new InstantCommand(() -> mCamera.switch(Constants.Camera.kFrontId)));
+                new InstantCommand(() -> mCamera.switch(Constants.Camera.kFrontId)));
```
Comment (omentic) [inline src/main/java/com/spartronics4915/frc2020/RobotContainer.java:138]: Indentation

File: src/main/java/com/spartronics4915/frc2020/RobotContainer.java [hunk 5]
```diff
@@ -128,17 +135,18 @@ private void configureJoystickBindings()
...
-            new InstantCommand(() -> mCamera.switch(Constants.Camera.kIntakeId)));
+                new InstantCommand(() -> mCamera.switch(Constants.Camera.kIntakeId)));
         new JoystickButton(mJoystick, 11).whenPressed(
-            new InstantCommand(() -> mCamera.switch(Constants.Camera.kTurretId)));
+                new InstantCommand(() -> mCamera.switch(Constants.Camera.kTurretId)));
         */
-        new JoystickButton(mJoystick, 1).toggleWhenPressed(new ShootBallTest(new Launcher()));
-        new JoystickButton(mJoystick, 7).whileHeld(new TrajectoryTrackerCommand(mDrive,
-                throughTrench(), mRamseteController, mStateEstimator.getCameraRobotStateMap()));
+        //new JoystickButton(mJoystick, 1).toggleWhenPressed(mLauncherCommands.new ShootBallTest(mLauncher));
```
Comment (omentic) [inline src/main/java/com/spartronics4915/frc2020/RobotContainer.java:146]: Uncomment these out

File: src/main/java/com/spartronics4915/frc2020/RobotContainer.java [hunk 6]
```diff
@@ -155,10 +163,8 @@ private void configureButtonBoardBindings()
         new JoystickButton(mButtonBoard, 5).whenPressed(new LauncherCommands.AimHigh(mLauncher));
         */
 
-        new JoystickButton(mButtonBoard, 6)
-            .whenPressed(mPanelRotatorCommands.new Raise(mPanelRotator));
-        new JoystickButton(mButtonBoard, 7)
-            .whenPressed(mPanelRotatorCommands.new Lower(mPanelRotator));
+        new JoystickButton(mButtonBoard, 6).whenPressed(mPanelRotatorCommands.new Raise(mPanelRotator));
```
Comment (omentic) [inline src/main/java/com/spartronics4915/frc2020/RobotContainer.java:166]: Revert indentation

File: src/main/java/com/spartronics4915/frc2020/commands/LauncherCommands.java [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+
+import com.spartronics4915.frc2020.subsystems.Launcher;
+
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import edu.wpi.first.wpilibj2.command.FunctionalCommand;
+
+public class LauncherCommands
+{
+    /*
```
Comment (omentic) [inline src/main/java/com/spartronics4915/frc2020/commands/LauncherCommands.java:11]: Turn into documentation comment syntax

File: src/main/java/com/spartronics4915/frc2020/commands/LauncherCommands.java [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+        // Called once the command ends or is interrupted.
+        @Override
+        public void end(boolean interrupted)
+        {
+            mLauncher.reset();
+        }
+    }
+
+
+    /*Default command of the launcher subsystem, makes the flywheel's target rps 0*/
```
Comment (omentic) [inline src/main/java/com/spartronics4915/frc2020/commands/LauncherCommands.java:99]: Syntax

File: src/main/java/com/spartronics4915/frc2020/subsystems/Launcher.java [hunk 9]
```diff
@@ -18,38 +22,49 @@
     private Servo mAngleAdjusterMasterServo;
```
Comment (omentic) [inline src/main/java/com/spartronics4915/frc2020/subsystems/Launcher.java:22]: Make the hardware `final`

File: src/main/java/com/spartronics4915/frc2020/subsystems/Launcher.java [hunk 10]
```diff
@@ -18,38 +22,49 @@
...
+                logInitialized(false);
+            } else {
+                logInitialized(true);
+            }
+            mFlywheelEncoder = mFlywheelMasterMotor.getEncoder();
+            
             // One NEO 550 motor for turret
             mTurretMotor = SpartronicsMax.makeMotor(Constants.Launcher.kTurretId, SensorModel.toRadians(360));
-            if (mFlywheelMasterMotor.hadStartupError() || mTurretMotor.hadStartupError())
+            if (mTurretMotor.hadStartupError())
```
Comment (omentic) [inline src/main/java/com/spartronics4915/frc2020/subsystems/Launcher.java:54]: Add `mFlywheelMasterMotor.hadStartupError()` (that's on me)

File: src/main/java/com/spartronics4915/frc2020/subsystems/Launcher.java [hunk 11]
```diff
@@ -18,38 +22,49 @@
...
-                Constants.Launcher.kFlywheelFollowerId);// new SpartronicsMax(Constants.Launcher.kFlywheelMasterID,SensorModel.toRadians(1));
+            mFlywheelMasterMotor = SpartronicsMax.makeMotor(/*Constants.Launcher.kFlywheelMasterId*/2, SensorModel.fromMultiplier(1));
+            mFlywheelMasterMotor.setVelocityGains(0.000389, 0, 0, 0);
+            mFeedforwardCalculator = new SimpleMotorFeedforward(Constants.Launcher.kS, Constants.Launcher.kV, Constants.Launcher.kA);
+            mFlywheelMasterMotor.setOutputInverted(true);
+            if (mFlywheelMasterMotor.hadStartupError())
+            {
+                mFlywheelMasterMotor = new SpartronicsSimulatedMotor();
+                logInitialized(false);
+            } else {
```
Comment (omentic) [inline src/main/java/com/spartronics4915/frc2020/subsystems/Launcher.java:47]: Syntax

File: src/main/java/com/spartronics4915/frc2020/subsystems/Launcher.java [hunk 12]
```diff
@@ -18,38 +22,49 @@
...
+            
             // One NEO 550 motor for turret
             mTurretMotor = SpartronicsMax.makeMotor(Constants.Launcher.kTurretId, SensorModel.toRadians(360));
-            if (mFlywheelMasterMotor.hadStartupError() || mTurretMotor.hadStartupError())
+            if (mTurretMotor.hadStartupError())
             {
-                mFlywheelMasterMotor = new SpartronicsSimulatedMotor();
                 mTurretMotor = new SpartronicsSimulatedMotor();
                 logInitialized(false);
             } else {
```
Comment (omentic) [inline src/main/java/com/spartronics4915/frc2020/subsystems/Launcher.java:58]: Syntax

File: src/main/java/com/spartronics4915/frc2020/subsystems/Launcher.java [hunk 13]
```diff
@@ -18,38 +22,49 @@
     private Servo mAngleAdjusterMasterServo;
     private Servo mAngleAdjusterFollowerServo;
     private SpartronicsMotor mTurretMotor;
-    private AnalogPotentiometer mTurretPotentiometer;
-    private double targetRPM;
+    private SpartronicsEncoder mTurretPotentiometer;
```
Comment (omentic) [inline src/main/java/com/spartronics4915/frc2020/subsystems/Launcher.java:25]: Rename - `mTurretEncoder`

File: src/main/java/com/spartronics4915/frc2020/subsystems/Launcher.java [hunk 14]
```diff
@@ -18,38 +22,49 @@
...
                 logInitialized(true);
             }
-            mFlywheelEncoder = mFlywheelMasterMotor.getEncoder();
+            mTurretPotentiometer = mTurretMotor.getEncoder().setFeedBackDevice(((SpartronicsMax)mTurretMotor).getAnalog());//new AnalogPotentiometer(Constants.Launcher.kTurretPotentiometerId, 90, -45);
 
             // Two Servos for angle adjustement
             mAngleAdjusterMasterServo = new Servo(Constants.Launcher.kAngleAdjusterMasterId);
             mAngleAdjusterFollowerServo = new Servo(Constants.Launcher.kAngleAdjusterFollowerId);
-
+
```
Comment (omentic) [inline src/main/java/com/spartronics4915/frc2020/subsystems/Launcher.java:66]: Remove

File: src/main/java/com/spartronics4915/lib/hardware/motors/SpartronicsMax.java [hunk 15]
```diff
@@ -111,7 +113,14 @@ public SpartronicsEncoder getEncoder()
     {
         return mEncoder;
     }
-
+    /**
+	 * @return An object for interfacing with a connected analog sensor.
+	 */
+    
+	public CANAnalog getAnalog() {
```
Comment (omentic) [inline src/main/java/com/spartronics4915/lib/hardware/motors/SpartronicsMax.java:120]: Syntax

File: src/main/java/com/spartronics4915/lib/hardware/motors/SpartronicsMotor.java [hunk 16]
```diff
@@ -150,4 +150,6 @@
      * @return The output current of the motor in amps.
      */
     double getOutputCurrent();
+
```
Comment (omentic) [inline src/main/java/com/spartronics4915/lib/hardware/motors/SpartronicsMotor.java:153]: Spacing

**Comment by omentic** [comment]
Closing - being the one who created this PR, I can't edit files directly.

**Pull request closed without merging**